Edits to System Setup. (#936)
hadley/r-pkgs: setup.Rmd

```

Make sure you have a recent version of the RStudio integrated development environment (IDE).
New versions are released regularly, so we recommend updating often to get access to the latest and greatest features.

Download the current version of RStudio Desktop here: <https://posit.co/download/rstudio-desktop/>.

> Most readers can use the free, open source version of RStudio Desktop.

Reviewer: Yeah, this is a good edit. Just to provide you with more background, here's an excerpt from a related slack conversation, when we made similar changes to the pre-workshop setup instructions for WTF. These quotes are from folks in our support organization, confirming my sense that the preview version is much less relevant than in the past:

> ... the Preview release is usually the release candidate for the next general release. It’s not updated often - just before a new release, when the engineering team thinks they have a good candidate. By definition, the Preview release will match the production release until the following release has a candidate, usually a few weeks at most before the next release

> We are definitely releasing more often than we used to, and we therefore put out fewer Previews between GA releases (bigger, longer release cycles mean there’s more risk for regressions and bugs that require new Previews before GA). It’s true that even when Preview and GA differ, there isn’t usually a big gap anymore, with the exception of releases with breaking changes (though those are rare).

> Previews have been utilized less with the change to shorter release cycles. We may have a preview up after "feature complete" for a couple of weeks before a release while we stabilize/certify the final build and that's it.

## devtools, usethis, and you {#sec-setup-usage}

> "I am large, I contain multitudes."
>
> --- Walt Whitman, Song of Myself

> As mentioned in @sec-intro-phil, devtools is a 'meta-package', encompassing and exposing functionality maintained in seven smaller packages[^setup-1].

Reviewer: ```suggestion
As mentioned in @sec-intro-phil, devtools is a 'meta-package', encompassing and exposing functionality maintained in several smaller packages[^setup-1].
```

I have learned the hard way that vague wording holds up better 😬

Author: I'm putting that on the list of "life skills from Jenny" 😜

For example, devtools might provide a wrapper function in order to set user-friendly defaults, introduce helpful interactive behaviour, or to combine functionality from multiple sub-packages.

> In some cases it simply re-exports a function from another package to make it easily available when devtools is loaded.

Reviewer: ```suggestion
In some cases it simply re-exports a function from another package to make it easily available when devtools is attached.
```

Sort of a nitpick. We do play loose with load vs. attach in some places in the book, but mostly try to be pretty pedantic about it.

Author: Yes, great catch

> [^setup-1]: [remotes](https://remotes.r-lib.org/), [pkgbuild](https://github.com/r-lib/pkgbuild), [pkgload](https://github.com/r-lib/pkgload), [rcmdcheck](https://github.com/r-lib/rcmdcheck), [revdepcheck](https://github.com/r-lib/revdepcheck), [sessioninfo](https://github.com/r-lib/sessioninfo), [usethis](https://usethis.r-lib.org/)

Reviewer: ```suggestion
[^setup-1]: At the time of writing, devtools exposes functionality from [remotes](https://remotes.r-lib.org/), [pkgbuild](https://github.com/r-lib/pkgbuild), [pkgload](https://github.com/r-lib/pkgload), [rcmdcheck](https://github.com/r-lib/rcmdcheck), [revdepcheck](https://github.com/r-lib/revdepcheck), [sessioninfo](https://github.com/r-lib/sessioninfo), and [usethis](https://usethis.r-lib.org/)
```

More rewording to be a bit more future-proof. Shouldn't this include roxygen2 and testthat?

Author: Some of these links are to the GitHub repos, and some are to the pkgdown pages. I got the original list (and links) from the [conscious uncoupling](https://www.tidyverse.org/blog/2018/10/devtools-2-0-0/#conscious-uncoupling) blog, but that was a while ago now. Should they all be pkgdown now? On the [devtools pkgdown site](https://devtools.r-lib.org/#conscious-uncoupling) all of the links go to the repos...

Reviewer: Re: why the blog didn't list roxygen2 and testhat. Presumably that's because those two packages were already well-established and had a full life of their own at the time of the uncoupling. Versus being created by carving out functionality from devtools itself.

So I think, given how we're approaching the history here, it's appropriate to include roxygen2 and testthat in this list.

I think we should link to a pkgdown site when it exists. But it won't for some of the more low-level packages, in which case GitHub repo is fine.

Author: Perfect, thanks for that explanation - I'll update the links to pkgdown where available.

> What's our recommended approach to devtools and its constituent packages?
> It varies, depending on your intentions when using the functions:

Reviewer: ```suggestion
What's our recommended approach to using devtools and its constituent packages?
It varies, depending on your intention:
```

> -   If you are using the functions interactively to help you develop your package, you should think of devtools as the provider of your favorite functions for package development. In this case you should attach devtools and call the functions without qualification (e.g., `load_all()`).

Reviewer: ```suggestion
-   If you are using the functions interactively to help you develop your package, you should think of devtools as the provider of your favorite functions for package development. In this case you should attach devtools with `library(devtools)` and call the functions without qualification (e.g., `load_all()`).
```

devtools::install_github("r-lib/usethis")
```

> ## R build toolchain {#setup-tools}

Reviewer: I'm not sure we can entirely remove this. But in the name of merging this PR, which is overwhelmingly helpful, I can accept the deletion. But will you open an issue about re-inserting some suitable stub re: setting your machine for package development? And paste in the text that we are deleting?

Author: Will do - sorry I was a bit overzealous!

Author: Opened in #937
